### PR TITLE
ev11l78a: add en_sink as a custom binding.

### PR DESCRIPTION
**Pull request by Flameeyes**
ev11l78a: add en_sink as a custom binding.

This exposes the EN_SINK signal as a custom binding, since it is ore to
the business logic of the board.

EN_SINK is a GPIO line connected to a MOSFET that provides power on the
screw terminal of the board, and should be enabled after USB-PD
negotiation.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/zephyrproject-rtos/zephyr/pull/61740).
* __->__ #61740
* #62478
* #61797

### CODE REVIEW

File: boards/arm/ev11l78a/ev11l78a.dts [hunk 1]
```diff
@@ -41,6 +54,11 @@
 		sense-gain-mult = <100>;
 	};
 
+	enable_sink: enable_sink {
+		compatible = "microchip,ev11l78a-ensink";
+		gpios = <&porta 22 GPIO_ACTIVE_HIGH>;
+	};
```
Comment (gmarull) [inline boards/arm/ev11l78a/ev11l78a.dts:60]: looks like a `regulator-fixed`?

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/arm/ev11l78a/ev11l78a.dts [hunk 2]
```diff
@@ -32,6 +33,18 @@
...
+		ind_1_5a: led_1 {
+			gpios = <&porta 1 GPIO_ACTIVE_HIGH>;
+			label = "1.5A_IND";
+		};
+
+		ind_3_0a: led_2 {
+			gpios = <&porta 15 GPIO_ACTIVE_HIGH>;
+			label = "3.0A_IND";
+		};
 	};
```
[the remark this answers is not in the document]
Comment (Flameeyes) [reply boards/arm/ev11l78a/ev11l78a.dts:48]: It seemed the easiest way to expose these with an actual label — but I'm not convinced myself. What other options are there?

**Review by Flameeyes — COMMENTED** [review]
Let me move the important change (memory and flash size) to #61744 that is more generic, and I'll follow up here about the regulator/sink enable option.

File: boards/arm/ev11l78a/ev11l78a.dts (hunk 1 above)
Comment (gmarull) [reply boards/arm/ev11l78a/ev11l78a.dts:61]: so if controlled by TCPC it should be a property there, something like:

```
tcpc {
    enable-sink-gpios =  <>;
};
```

then TCPC code would use the GPIO API. Or, if using regulators:

```
enable_sink: enable-sink {
    compatible = "regulator-fixed";
    enable-gpios = <...>;
};

...

tcpc {
    ...
    sink-supply = <&enable_sink>;
};
```

Then TCPC would use the regulators API (main advantage is that is reference counted, useful if supply is shared).

Comment (gmarull) [reply boards/arm/ev11l78a/ev11l78a.dts:61]: It depends on how it is used, if by multiple clients then regulator is a good fit, is by a single client, a GPIO does the job. A typical case for GPIO usage is e.g. on sensors that have a shutdown pin. I don't know this case in detail to suggest one way or the other.

Comment (Flameeyes) [reply boards/arm/ev11l78a/ev11l78a.dts:61]: To be honest, this is almost the literal case from [the documentation you linked earlier](https://docs.zephyrproject.org/latest/samples/basic/custom_dts_binding/README.html) as it's a GPIO connected to a specific MOSFET…

Leaving it as a generic GPIO doesn't seem right though — it's most certainly no longer "General-purpose" once it's wired to the MOSFET on the board, and it has a very specific function.

**Comment by nandojve** [comment]
You need rebase : )

**Comment by nandojve** [comment]
Ping

**Comment by nandojve** [comment]
Hi @Flameeyes ,

Is there anything pending from your side on this?

**Pull request closed without merging**